### PR TITLE
Do not allow source suppression for VS/host diagnostics

### LINKED ISSUE #23775: AnalyzerDependencyCheckingService.CheckForConflictsAsync results in 15 seconds of blocked UI opening Roslyn
**Version Used**: 
Visual Studio 2017 version 15.5

**Steps to Reproduce**:
1. Open a large solution where there are analyzer diagnostics in a "leaf" project.

**Expected Behavior**:
Solution should open in a reasonable time

**Actual Behavior**:
It's very slow, because creating the diagnostic requires a full compilation to be built and then examined for source level suppressions, and this happens synchronously on the UI thread as part of solution load.

`microsoft.visualstudio.languageservices   <<microsoft.visualstudio.languageservices!Microsoft.VisualStudio.LanguageServices.Implementation.AnalyzerDependencyCheckingService+<CheckForConflictsAsync>d__14.MoveNext()>> | 11.2 | 9,363.414 | 4,087`

### PR DESCRIPTION
**Pull request by mavasani**
This fixes the UI delay from force binding all assembly attributes during solution load.
Fixes #23775

<details><summary>Ask Mode template</summary>

### Customer scenario

Customer loads a solution that has more than one analyzer with conflicting dependency contents. This leads to analyzer conflict diagnostics getting reported during solution load. Reporting these diagnostic force completes binding of all assembly attributes in the compilation on the UI thread, causing a very large UI delay.

### Bugs this fixes

#23775

### Workarounds, if any

N/A

### Risk

Low. We are not executing a bunch of code that force completes the attributes to check for source suppressions. Users are recommended to suppress analyzer conflict diagnostics through compilation options.

### Performance impact

This should improve solution load performance in presence of analyzer dependency conflict diagnostics.

### Is this a regression from a previous update?

No.

### Root cause analysis

This was a known issue for a while, but we thought it was pretty rare, and something that would go away with us moving to new project system doing this work on a background thread.

### How was the bug found?

Dogfooding.

### Test documentation updated?

N/A

</details>

### CODE REVIEW

File: src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs [hunk 1]
```diff
@@ -104,6 +104,35 @@ public abstract partial class Diagnostic : IEquatable<Diagnostic>, IFormattable
             IEnumerable<Location> additionalLocations,
             ImmutableDictionary<string, string> properties,
             params object[] messageArgs)
+        {
+            return Create(descriptor, location, effectiveSeverity: descriptor.DefaultSeverity, additionalLocations, properties, messageArgs: messageArgs);
```
Comment (Pilchie) [inline src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs]: Why use named args for `messageArgs`?

Comment (mavasani) [reply src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs]: Ah, that came from copying the code above. Will fix it.

**Comment by mavasani** [comment]
> @tmeschter suggested adding some telemetry in this case as well, is that something you want to do as part of this change?

We already log telemetry in analyzer dependency checking service: http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/AnalyzerDependency/AnalyzerDependencyCheckingService.cs,127

**Comment by mavasani** [comment]
@dotnet-bot retest windows_release_unit32_prtest please

**Comment by mavasani** [comment]
@Pilchie approved to merge?

**Comment by Pilchie** [comment]
Yes - was waiting for a second review yesterday.

**Pull request merged**